src/functions/marshal_uri_from_sapi.php

```diff
@@ -138,45 +126,39 @@ function marshalUriFromSapi(array $server, array $headers): Uri
      *
      * From Laminas\Http\PhpEnvironment\Request class
      */
-    $marshalRequestPath = function (array $server): string {
+    $marshalRequestPath = static function (array $server): string {
         // IIS7 with URL Rewrite: make sure we get the unencoded url
         // (double slash problem).
         $iisUrlRewritten = $server['IIS_WasUrlRewritten'] ?? null;
         $unencodedUrl    = $server['UNENCODED_URL'] ?? '';
         if ('1' === $iisUrlRewritten && ! empty($unencodedUrl)) {
             return $unencodedUrl;
         }
-
         $requestUri = $server['REQUEST_URI'] ?? null;
-
         if ($requestUri !== null) {
             return preg_replace('#^[^/:]+://[^/]+#', '', $requestUri);
         }
-
         $origPathInfo = $server['ORIG_PATH_INFO'] ?? null;
         if (empty($origPathInfo)) {
             return '/';
         }
-
         return $origPathInfo;
     };
 
     $uri = new Uri('');
 
     // URI scheme
     $scheme            = 'http';
-    $marshalHttpsValue = function ($https): bool {
+    $marshalHttpsValue = static function ($https): bool {
         if (is_bool($https)) {
             return $https;
         }
-
```

Comment:
Not sure why your CS adjustments removed all these spaces: seems invalid?

Makes the entire thing much less readable too...

Reply:
Yes, that's known drawback of using rector https://github.com/rectorphp/rector#known-drawbacks , I can rollback the lines if needed.

Reply:
Yes please: I know it's annoying, but some of these spacing changes do need adjustment indeed :)

Reply:
I rolled back spaces https://github.com/laminas/laminas-diactoros/pull/115/commits/cf8ec668d3f7c189fb578a32992f488914b3603a